### ISSUE TITLE
Kubernetes failures can cause fatal errors

### ISSUE THREAD

**Issue by XAMPPRocky** [googleforgames/quilkin#697]
I've seen the following (separate) errors in running the latest release of the control plane. These errors should be prevented from becoming fatal for the control plane, the control plane should be catching this error and retrying rather than terminating.

```
watch stream failed: Error reading events stream: error reading a body from connection: error reading a body from connection: Connection reset by peer (os error 104)
```

```
error returned by apiserver during watch: too old resource version: 53056602 (53056699): Expired
```

```
failed to perform initial object list: ApiError: \"404 page not found\\n\": Failed to parse error data (ErrorResponse { status: \"404 Not Found\", message: \"\\\"404 page not found\\\\n\\\"\", reason: \"Failed to parse error data\", code: 404 })
```

**Comment by markmandel** [comment]
Do you know where this is originating from? I could take a look if you haven't got to it already.

**Comment by XAMPPRocky** [comment]
> Do you know where this is originating from? I could take a look if you haven't got to it already.

These are happening in the Agones discovery task. It might be fixed by the changes in #662, but I haven't tested that. The fix should just be that the Agones' task retries when it encounters one of these errors.

**Comment by markmandel** [comment]
Is it worth just pulling the fix out of #662 so we don't have to wait for the whole feature?
